### PR TITLE
fix: ensure `--build-output` destination exists

### PR DESCRIPTION
**Pull request by szymonrybczak**
## Summary:

Replicates https://github.com/react-native-community/cli/commit/48d4c29bba4e8b16cbc8307bd1b4c5349f3651d8, which landed inside `cli-plugin-metro` inside RNC CLI, but because of migration of code to `community-cli-plugin` it looks like apparently the fix wasn't replicated. 

## Changelog:

[GENERAL] [FIXED] - Ensure `--build-output` destination exists

## Test Plan:

Specify a new directory that doesn't exists inside `--build-output`:

`npx react-native bundle --build-output dist/new-dir/index.bundle`

and this command shouldn't fail.

### CODE REVIEW

**Comment by szymonrybczak** [comment]
cc @tido64 just FYI

File: packages/community-cli-plugin/src/commands/bundle/buildBundle.js [hunk 1]
```diff
@@ -112,6 +113,12 @@ async function buildBundleWithConfig(
   try {
     const bundle = await bundleImpl.build(server, requestOpts);
 
+    // Ensure destination directory exists before saving the bundle
+    fs.mkdirSync(path.dirname(args.bundleOutput), {
+      recursive: true,
+      mode: 0o755,
+    });
```
Comment (satya164) [inline packages/community-cli-plugin/src/commands/bundle/buildBundle.js:120]: Maybe async?

Comment (szymonrybczak) [reply packages/community-cli-plugin/src/commands/bundle/buildBundle.js]: https://github.com/facebook/react-native/pull/45182/commits/a75be017ccf8982e7a10fbc8b25d818e9eedfcd0

File: packages/community-cli-plugin/src/commands/bundle/buildBundle.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ import parseKeyValueParamArray from '../../utils/parseKeyValueParamArray';
 import saveAssets from './saveAssets';
 import {logger} from '@react-native-community/cli-tools';
 import chalk from 'chalk';
+import fs from 'fs/promises';
```
Comment (huntie) [inline packages/community-cli-plugin/src/commands/bundle/buildBundle.js:21]: Please adjust this import to fix the Flow error.

image

```suggestion
import {promises as fs} from 'promises';
```

**Pull request closed without merging**